shared/src/main/kotlin/org/javacs/kt/classpath/GradleClassPathResolver.kt

```diff
@@ -10,34 +12,65 @@ import java.nio.file.Files
 import java.nio.file.Path
 import java.nio.file.Paths
 
-internal class GradleClassPathResolver(private val path: Path, private val includeKotlinDSL: Boolean): ClassPathResolver {
+internal class GradleClassPathResolver(private val path: Path, private val includeKotlinDSL: Boolean, private val storage: Storage?): ClassPathResolver {
     override val resolverType: String = "Gradle"
-    private val projectDirectory: Path get() = path.getParent()
+    private val projectDirectory: Path get() = path.parent
+
+    private var cachedClassPath: Set<ClassPathEntry>? = storage?.getObject("gradleClassPath")
```

Comment:
Just an idea, not sure how feasible that would be, could we have a uniform class path file that caches the entire class path (independent of whether Maven or Gradle was originally used to resolve it) and implement that as a new `ClassPathResolver` that wraps the existing ones (e.g. something along the lines of `CachingClassPathResolver`)? That way we wouldn't have to include caching in every resolver implementation separately. Feel free to correct me if I'm missing anything though. Perhaps detecting changes would become quite a bit harder that way.

Reply:
This is a very nice idea. I'll give it a try 👍